src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.Intrinsics.cs

```diff
@@ -179,7 +179,7 @@ private CorInfoIntrinsics getIntrinsicID(MethodDesc method, byte* pMustExpand)
             if (pMustExpand != null)
                 *pMustExpand = 0;
 
-            Debug.Assert(method.IsIntrinsic);
+            Debug.Assert(method.IsIntrinsic || method.IsInternalCall);
```

Comment:
This should be reverted.

Reply:
Reverted in 3rd commit, thanks for pointing that out!